### ISSUE TITLE
tmpfs-mode does not default to 1777 as documented

### ISSUE THREAD

**Issue by jdmarshall** [docker/docs#15594]
File: [storage/tmpfs.md](https://docs.docker.com/storage/tmpfs/) reports that the default mode is 1777. I don't know when this was the case, but it's not true in 20.10.7

docker --version
Docker version 20.10.7, build f0df350

```
docker run -it --mount type=tmpfs,destination=/var/lib/nginx/tmp my-image:latest /bin/bash

ls -la /var/lib/nginx

...
drwx------    2 root     root            40 Sep  7 18:56 tmp
```

In fact tmpfs-mode seems to be ignored if the file being mounted already exists:

```
docker run -it --mount type=tmpfs,destination=/var/lib/nginx/tmp,tmpfs-mode=1777 my-image:latest /bin/bash
bash-5.1# ls -la /var/lib/nginx
...
drwx------    2 root     root            40 Sep  7 19:02 tmp
```

Note that originally the tmp directory was nginx:nginx, so this breaks things quite badly.

Near as I can tell, this set of instructions only functions if the mount point doesn't exist at all on the image, which is not a manner of running or building Docker images that I've noticed anyone using before. It seems to me that clobbering a directory with a mount is the default, not a corner case.

**Comment by jdmarshall** [comment]
A workaround for this issue would be nice. I'm running into problems with it again.

**Comment by dvdksn** [comment]
🧟 It seems to have been fixed (?) between docker 20.10 and now. 

```
$ docker run -it --mount type=tmpfs,destination=/var/lib/nginx/tmp debian:latest /bin/bash
root@950005820c5f:/# ls -la /var/lib/nginx
total 8
drwxr-xr-x 3 root root 4096 Dec 20 14:24 .
drwxr-xr-x 1 root root 4096 Dec 20 14:24 ..
drwxrwxrwt 2 root root   40 Dec 20 14:24 tmp
```

Feel free to open another issue if you encounter another issue.

**Comment by jdmarshall** [comment]
Thanks for the heads up.
